src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -159,6 +159,8 @@ def load_arguments(self, _):
             c.argument('enable_bursting', arg_type=get_three_state_flag(), help='Enable on-demand bursting beyond the provisioned performance target of the disk. On-demand bursting is disabled by default, and it does not apply to Ultra disks.')
             c.argument('public_network_access', arg_type=get_enum_type(['Disabled', 'Enabled']), min_api='2021-04-01', is_preview=True, help='Customers can set on Managed Disks or Snapshots to control the export policy on the disk.')
             c.argument('accelerated_network', arg_type=get_three_state_flag(), min_api='2021-04-01', is_preview=True, help='Customers can set on Managed Disks or Snapshots to enable the accelerated networking if the OS disk image support.')
+            if self.supported_api_version(min_api='2021-12-01', operation_group='disks') and self.supported_api_version(min_api='2021-12-01', operation_group='snapshots'):
+                c.argument('architecture', arg_type=get_enum_type(self.get_models('Architecture', operation_group='disks')), help='CPU architecture.')
```

Comment:
It is suggested to split the registration of `architecture` for `disk` and `snapshot` into two independent parts, so as to avoid coupling